### PR TITLE
Move experimental stuff to `nk.experimental`

### PR DESCRIPTION
**Pull request by PhilipVinc**
@femtobit this looks good?

I have a few people who are using `nk.vqs.experimental.***` so I deprecated it.

This does not explicitly import `nk.experimental` until somebody does explicitly `from netket import experimental`, at which point it became visible.
We can also make it visible by default.

### CODE REVIEW

**Comment by PhilipVinc** [comment]
on second thought I'd prefer to have this available by default.
I hate it on tax that you have to import stuff explicitly.

**Comment by femtobit** [comment]
>  I hate it on tax that you have to import stuff explicitly.

Actually, in case of `experimental` I prefer your first version, i.e., requiring an explicit import.

**Comment by PhilipVinc** [comment]
my poor PR title!

**Comment by PhilipVinc** [comment]
> Actually, in case of experimental I prefer your first version, i.e., requiring an explicit import.

I'm fine with both.
However I'm curious, why do you prefer this one if I may?

**Comment by femtobit** [comment]
I mean, in other places you've introduced the requirement to set `NETKET_EXPERIMENTAL=1` in the environment. I feel compared to that, the explicit import is not too taxing.

Also, as a minor point, I like to do something like
```python
import netket as nk
import netket.experimental as nkx
```
which makes using `nkx` quite concise (but still explicit).

**Comment by femtobit** [comment]
> my poor PR title!

I'd rather change it now than forget it before merging the commit with this title. But I do indeed like the PR. :stuck_out_tongue:

**Comment by PhilipVinc** [comment]
fair point.

2 more questions:
 - shall we move ParallelTempering algorithms here too? the also require to set the env variable right now. (I'd say yes...)
 - where do I put this stuff in docs now? a new experimental API page?

**Comment by femtobit** [comment]
I;d say yes to both of your questions.

Regarding the need for the environment variable: I think explicitly requiring users to import features from `netket.experimental`, in particular with it not being loaded by default, is enough to make it clear that this is indeed experimental and may change in the future. I don't really see a need for an additional opt-in through the env variable. But I can live with it too, if you prefer too keep it.

**Comment by PhilipVinc** [comment]
>Regarding the need for the environment variable: I think explicitly requiring users to import features from netket.experimental, in particular with it not being loaded by default, is enough to make it clear that this is indeed experimental and may change in the future. I don't really see a need for an additional opt-in through the env variable. But I can live with it too, if you prefer too keep it.

Nono. I'd love to get rid of this env variable.
I just had no better idea at the time

**Comment by PhilipVinc** [comment]
https://www.netket.org/preview/pv/experimental/docs/api-experimental.html

**Comment by PhilipVinc** [comment]
ok, I think it's done?

File: CHANGELOG.md [hunk 1]
```diff
@@ -14,7 +14,10 @@
 
 * In addition to nearest-neighbor edges, `Lattice` can now generate edges between next-nearest and, more generally, k-nearest neighbors via the constructor argument `max_neighbor_order`. The edges can be distinguished by their `color` property (which is used, e.g., by `GraphOperator` to apply different bond operators). [#970](https://github.com/netket/netket/pull/970)
 
+* A new `netket.experimental` module (hidden by default) has been created. All experimental features have been moved there. [#976](https://github.com/netket/netket/pull/976)
```
Comment (femtobit) [inline CHANGELOG.md:17]: Maybe this is a bit more clear:
```suggestion
* A new `netket.experimental` submodule has been created and all experimental features have been moved there. Note that in contrast to the other `netket` submodules, `netket.experimental` is not imported by default.  [#976](https://github.com/netket/netket/pull/976)
```

**Review by femtobit — APPROVED** [review]
Just one suggestion, otherwise looks great to me!

**Pull request merged**